rocky/reports/forms.py

```diff
@@ -4,22 +4,21 @@
 from django.utils.translation import gettext_lazy as _
 from tools.forms.base import BaseRockyForm
 
+from octopoes.models.types import OOIType
 from reports.report_types.definitions import Report
-from reports.report_types.helpers import get_ooi_types_with_report
-
-SORTED_OOI_TYPES_FOR_REPORT = sorted([ooi_class.get_ooi_type() for ooi_class in get_ooi_types_with_report()])
-
-OOI_TYPE_CHOICES_FOR_REPORT = ((ooi_type, ooi_type) for ooi_type in SORTED_OOI_TYPES_FOR_REPORT)
 
 
 class OOITypeMultiCheckboxForReportForm(BaseRockyForm):
     ooi_type = forms.MultipleChoiceField(
         label=_("Filter by OOI types"),
         required=False,
-        choices=OOI_TYPE_CHOICES_FOR_REPORT,
         widget=forms.CheckboxSelectMultiple,
     )
 
+    def __init__(self, ooi_types: list[OOIType], *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["ooi_type"].choices = ((ooi_type, ooi_type) for ooi_type in ooi_types)
```

Comment:
Why? Another form class (possibly overriding this one) is imho a better and cleaner solution

Also use `List` instead of `list`

Reply:
the ooi types are fetched based on the reports, it will then initiate only those ooi types choices that are listed in te reports